src/main/java/com/faforever/client/leaderboard/LeaderboardController.java

```diff
@@ -0,0 +1,369 @@
+
+import java.nio.file.Path;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+
+@Slf4j
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@RequiredArgsConstructor
+public class LeaderboardController implements Controller<Tab> {
+
+  private static final PseudoClass NOTIFICATION_HIGHLIGHTED_PSEUDO_CLASS = PseudoClass.getPseudoClass("highlighted-bar");
+
+  private final AssetService assetService;
+  private final I18n i18n;
+  private final LeaderboardService leaderboardService;
+  private final NotificationService notificationService;
+  private final PlayerService playerService;
+  private final UiService uiService;
+  public Tab leaderboardRoot;
+  public TextField searchTextField;
+  public Pane connectionProgressPane;
+  public Pane contentPane;
+  public BarChart<String, Integer> ratingDistributionChart;
+  public Label playerDivisionNameLabel;
+  public Label playerScoreLabel;
+  public Label scoreLabel;
+  public Label seasonLabel;
+  public ComboBox<SubdivisionBean> majorDivisionPicker;
+  public Arc scoreArc;
+  public TabPane subDivisionTabPane;
+  public ImageView playerDivisionImageView;
+  public CategoryAxis xAxis;
+  public NumberAxis yAxis;
+  public Label placementLabel;
+  public Label seasonDateLabel;
+  private LeagueSeasonBean season;
+  private LeagueEntryBean entryToSelect;
+  private InvalidationListener playerRatingListener;
+
+  @VisibleForTesting
+  protected AutoCompletionBinding<String> usernamesAutoCompletion;
+
+  @Override
+  public void initialize() {
+    scoreLabel.setText(i18n.get("leaderboard.score").toUpperCase());
+    searchTextField.setPromptText(i18n.get("leaderboard.searchPrompt").toUpperCase());
+
+    JavaFxUtil.bindManagedToVisible(contentPane);
+    JavaFxUtil.bindManagedToVisible(connectionProgressPane);
+    connectionProgressPane.visibleProperty().bind(contentPane.visibleProperty().not());
+
+    majorDivisionPicker.setConverter(divisionStringConverter());
+    yAxis.setTickLabelFormatter(integerStringConverter());
+
+    searchTextField.textProperty().addListener((observable, oldValue, newValue) ->
+        processSearchInput(newValue));
+
+    subDivisionTabPane.widthProperty().addListener((observable, oldValue, newValue) ->
+        setTabWidth((double) newValue, subDivisionTabPane.getTabs().size()));
+  }
+
+  private void setTabWidth(double tabPaneWidth, int tabNumber) {
+    // The tabs always appear 40px wider, maybe because of some obscure padding rule.
+    // We also need to subtract the tab area padding to prevent horizontal scrolling.
+    double tabWidth = Math.floor((tabPaneWidth - 20.0) / tabNumber) - 40.0;
+    JavaFxUtil.runLater(() -> {
+      subDivisionTabPane.setTabMinWidth(tabWidth);
+      subDivisionTabPane.setTabMaxWidth(tabWidth);
+    });
+  }
```

Comment:
Because I want the tabs to always be in total the width of the tabpane. Sometimes it is five, sometimes only one. And generally the number could be arbitrary.

Reply:
Ah gotcha I thought this was setting the width of the tabpane